pandas/core/groupby/generic.py

```diff
@@ -1396,9 +1396,7 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
 
         op = GroupByApply(self, func, args=args, kwargs=kwargs)
         result = op.agg()
-        if not is_dict_like(func) and result is not None:
-            return result
```

Comment:
Can we just call `.reset_index()` when `as_index=False` here?

Reply:
I think you are right @rhshadrach, this seems to be the right way to go. However, simply doing this is failing certain test cases. I will try to investigate later.

Reply:
Update: according to my investigation, it needs to (at least) be something like
```python
return result if self.as_index or not is_list_like(func) else result.reset_index()
```
I will modify the PR in this way for now.